Software/lib/GNSE_HAL/GNSE_hal.c

```diff
@@ -0,0 +1,240 @@
+/** Copyright © 2021 The Things Industries B.V.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+/**
+ * @file GNSE_hal.c
+ *
+ * @copyright Copyright (c) 2021 The Things Industries B.V.
+ *
+ */
+
+#include "GNSE_hal.h"
+#include "app.h"
+#include "GNSE_acc.h"
+#include "GNSE_bm.h"
+
+/**
+  * @brief Deinitialises the Sensors for GNSE: the accelerometer and humidity/temperature sensor
+  * @note The I2C interface is not turned off by this function
+  * @param none
+  * @return none
+  */
+void GNSE_Sensors_DeInit(void)
+{
+    if (GNSE_ACC_DeInit() != ACC_OP_SUCCESS)
+    {
+#if (APP_LOG_ENABLED)
```

Comment:
This line has multiple issues:
1. You don't need #if every time you print, see this [line](https://github.com/TheThingsIndustries/generic-node-se/blob/9c839fa073ffa4921c7678ac52f23fa758c8678b/Software/app/app.h#L53) to understand why
2. App layer APIs should not be used here in libs, see how the [flash lib prints](https://github.com/TheThingsIndustries/generic-node-se/blob/9c839fa073ffa4921c7678ac52f23fa758c8678b/Software/lib/MX25R1635/mx_define.h#L34) but there is no need for this here.
3. Lib functions should not be as vocal as this

Please fix all following logging.